src/libmain.c

```diff
@@ -765,16 +769,20 @@ static gint setup_config_dir(void)
 	return mkdir_result;
 }
 
-/* Signal handling removed since main_quit() uses functions that are
- * illegal in signal handlers
-static void signal_cb(gint sig)
+
+#ifdef G_OS_UNIX
+static gboolean signal_cb(gpointer user_data)
 {
+	gint sig = GPOINTER_TO_INT(user_data);
 	if (sig == SIGTERM)
 	{
+		geany_debug("Received SIGTERM signal");
 		main_quit();
 	}
+	return G_SOURCE_REMOVE;
```

Comment:
I wasn't sure if this should be `G_SOURCE_REMOVE` or `G_SOURCE_CONTINUE`, I guess it doesn't matter since `SIGTERM` is the only signal handled here at this point and the application will exit anyway, but in the future we might like to change the return value depending what signal was handled and what action was taken.